pkg/compactor/block_upload.go

```diff
@@ -504,6 +509,33 @@ func (c *MultitenantCompactor) validateBlock(ctx context.Context, blockID ulid.U
 		}
 	}
 
+	// check that the blocks doesn't contain down-sampled data
+	if blockMetadata.Thanos.Downsample.Resolution > 0 {
+		return errors.New("block contains downsampled data")
+	}
```

Comment:
We can do this check much earlier -- when upload is starting. During `StartBlockUpload` we use `sanitizeMeta` method to check `meta.json` file before storing it to the bucket. We can check for resolution then, before any other files are uploaded.

Reply:
yes, nice catch, I've moved the check to `sanitizeMeta` and the corresponding unit test